src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_adapter.py

```diff
@@ -0,0 +1,428 @@
+from ...utils import (
+    PIL_INTERPOLATION,
+    is_accelerate_available,
+    is_accelerate_version,
+    logging,
+    replace_example_docstring,
+)
+from . import StableDiffusionPipelineOutput
+from .pipeline_stable_diffusion import StableDiffusionPipeline
+from .safety_checker import StableDiffusionSafetyChecker
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> from PIL import Image
+        >>> from diffusers.utils import load_image
+
+        >>> image = load_image("https://huggingface.co/RzZ/sd-v1-4-adapter-color/resolve/main/color_ref.png")
+
+        >>> color_palette = image.resize((8, 8))
+        >>> color_palette = color_palette.resize((512, 512), resample=Image.Resampling.NEAREST)
+
+        >>> import torch
+        >>> from diffusers import StableDiffusionAdapterPipeline, Adapter
+
+        >>> adapter = Adapter.from_pretrained("RzZ/sd-v1-4-adapter-color")
+        >>> pipe = StableDiffusionAdapterPipeline.from_pretrained(
+        ...     "RzZ/sd-v1-4-adapter",
+        ...     adapter=adapter,
+        ...     torch_dtype=torch.float16,
+        ... )
+
+        >>> pipe.to("cuda")
+
+        >>> out_image = pipe(
+        ...     "At night, glowing cubes in front of the beach",
+        ...     image=color_palette,
+        ...     generator=generator,
+        ... ).images[0]
+        ```
+"""
+
+
+def preprocess(image):
+    if isinstance(image, torch.Tensor):
+        return image
+    elif isinstance(image, PIL_Image.Image):
+        image = [image]
+
+    multi_control = isinstance(image[0], list) or isinstance(image[0], tuple)
+    multi_control |= isinstance(image[0], torch.Tensor) and image[0].ndim == 4
+
+    if multi_control:
+        images = [preprocess(subset) for subset in image]
+        b, c, h, w = images[0].shape
+        image = [img.reshape([b * c, h, w]) for img in images]
+
+    if isinstance(image[0], PIL_Image.Image):
+        w, h = image[0].size
+        w, h = map(lambda x: x - x % 8, (w, h))  # resize to integer multiple of 8
+
+        image = [np.array(i.resize((w, h), resample=PIL_INTERPOLATION["lanczos"]))[None, :] for i in image]
+        image = np.concatenate(image, axis=0)
+        image = np.array(image).astype(np.float32) / 255.0
+        image = image.transpose(0, 3, 1, 2)
+        # NOTE: offical adapter weight is trained with [0, 1] value space
+        # image = 2.0 * image - 1.0
+        image = torch.from_numpy(image)
+    elif isinstance(image[0], torch.Tensor):
+        if image[0].ndim == 3:
+            image = torch.stack(image, dim=0)
+        elif image[0].ndim == 4:
+            image = torch.cat(image, dim=0)
+        else:
+            raise ValueError(
+                f"Invalid image tensor! Expecting image tensor with 3 or 4 dimension, but recive: {image[0].ndim}"
+            )
+    return image
+
+
+class StableDiffusionAdapterPipeline(StableDiffusionPipeline):
```

Comment:
We unfortunately don't have pipelines inherit from other pipelines. All pipelines inherit from DiffusionPipeline. Common components can be shared through our `Copied from` mechanism. See this section here https://github.com/huggingface/diffusers/blob/main/docs/source/en/conceptual/philosophy.mdx#pipelines

Lemme know if you need help with this part :)

Reply:
Fixed